src/main/groovy/org/gradle/android/workarounds/room/AnnotationProcessorWorkaround.groovy

```diff
@@ -26,4 +26,18 @@ abstract class AnnotationProcessorWorkaround<T> {
     abstract void initWorkaround()
 
     abstract void configureWorkaroundTask(Task task)
+
+    boolean kspIsAppliedWithRoom() {
+        if (project.configurations.find { it.name == "ksp" } != null) {
+            boolean found = false
+            project.configurations.findByName("ksp").dependencies.forEach {
```

Comment:
Does this cause eager dependency resolution?
Is it okay to call this multiple times if it is a slow call?

Reply:
added `named` instead of `findByName` here https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/0233c5f1f185b50721de37598d54cfecd7908be6

In terms of performance, 
KaptWorkaround:
* Configuration: https://ge.solutions-team.gradle.com/s/47cgoojgorh3a/performance/configuration
* Dependency Resolution: https://ge.solutions-team.gradle.com/s/47cgoojgorh3a/performance/dependency-resolution

KspWorkaround:
* Configuration: https://ge.solutions-team.gradle.com/s/ktmuc4g332tmu/performance/configuration
* Dependency Resolution: https://ge.solutions-team.gradle.com/s/4547zpce43l6k/performance/dependency-resolution

In this last point is where we have the resolution of dependencies during configuration(Gradle initiated):
<img width="530" alt="Screenshot 2023-02-08 at 6 31 21 PM" src="https://user-images.githubusercontent.com/475733/217702512-a9221d74-c13a-4381-9f64-0eecc39c2479.png">

What do you think?

Reply:
We should avoid resolving dependencies during configuration time at all costs. I'm surprised that this is marked as Gradle initiated though.
The time shown is only the time to resolve the dependency but not to download them. This could add quite a lot of configuraiton time if we need to download them.

Reply:
tested again:

With Kotlin 1.8.10 and Ksp 1.8.10-1.0.9  there is a resolution:
https://ge.solutions-team.gradle.com/s/2jdicgzdkssd2/performance/dependency-resolution?toggled=dependency-resolution-configuration-gradle-initiated

If we execute the same task without the plugin we have the same result:
https://ge.solutions-team.gradle.com/s/o3mw7fivgfeeu/performance/dependency-resolution?toggled=dependency-resolution-configuration-gradle-initiated

The issue has been discussed here: https://github.com/google/ksp/issues/1294 and it's already merged.

This issue is not reproducible in KSP versions < 1.8.10

Reply:
Gotcha, when you said when testing without the plugin, you mean without the Android cache fix plugin, correct?

Sounds good then. 👍